libs/blocks/article-header/article-header.js

```diff
@@ -0,0 +1,198 @@
+import { createTag, getMetadata, getConfig } from '../../utils/utils.js';
+import { loadTaxonomy, getLinkForTopic, getTaxonomyModule } from '../article-feed/article-helpers.js';
+import { replaceKey } from '../../features/placeholders.js';
+import { fetchIcons } from '../../features/icons.js';
+import { buildFigure } from '../figure/figure.js';
+
+async function validateAuthorUrl(url) {
+  if (!url) return null;
+
+  const resp = await fetch(`${url}.plain.html`);
+  if (!resp || !resp.ok) {
+    console.log(`Could not retrieve metadata for ${url}`);
+    return null;
+  }
+
+  const html = await resp.text();
+  const parser = new DOMParser();
+  const doc = parser.parseFromString(html, 'text/html');
+
+  return doc;
+}
+
+function openPopup(e) {
+  const target = e.target.closest('a');
+  const href = target.getAttribute('data-href');
+  const type = target.getAttribute('data-type');
+  window.open(
+    href,
+    type,
+    'popup,top=233,left=233,width=700,height=467',
+  );
+}
+
+async function buildAuthorInfo(authorEl, bylineContainer) {
+  const { href, textContent } = authorEl;
+
+  const config = getConfig();
+  const base = config.miloLibs || config.codeRoot;
+  const authorImg = createTag('div', { class: 'article-author-image' });
+  authorImg.style.backgroundImage = `url(${base}/blocks/article-header/adobe-logo.svg)`;
+  bylineContainer.prepend(authorImg);
+
+  const doc = await validateAuthorUrl(href);
+  if (!doc) {
+    const p = createTag('p', null, textContent);
+    authorEl.replaceWith(p);
+    return;
+  }
+
+  const img = doc.querySelector('img');
+  if (img) {
+    img.setAttribute('alt', authorEl.textContent);
+    authorImg.append(img);
+    if (!img.complete) {
+      img.addEventListener('load', () => {
+        authorImg.style.backgroundImage = 'none';
+      });
+      img.addEventListener('error', () => {
+        img.remove();
+      });
+    } else {
+      authorImg.style.backgroundImage = 'none';
+    }
+  }
+}
+
+async function copyToClipboard(button) {
```

Comment:
I wonder if this should be moved into a central file.  Something like /libs/utils/tools.js
```suggestion
async function copyToClipboard(button, copyTxt) {
```